File: src/weaponinventorymod/gui/WimGuiListBounds.java
package weaponinventorymod.gui;

import com.fs.starfarer.api.input.InputEventAPI;
import com.fs.starfarer.api.ui.CustomPanelAPI;

final class WimGuiListBounds {
    private final int maxScrollOffset;
    private final float left;
    private final float top;
    private final float width;
    private final float height;

    WimGuiListBounds(int maxScrollOffset, float left, float top, float width, float height) {
        this.maxScrollOffset = maxScrollOffset;
        this.left = left;
        this.top = top;
        this.width = width;
        this.height = height;
    }

    int getMaxScrollOffset() {
        return maxScrollOffset;
    }

    boolean contains(CustomPanelAPI root, InputEventAPI event) {
        if (root == null || root.getPosition() == null || event == null || maxScrollOffset <= 0) {
            return false;
        }
        float screenLeft = root.getPosition().getX() + left;
        float screenRight = screenLeft + width;
        float screenTop = root.getPosition().getY() + root.getPosition().getHeight() - top;
        float screenBottom = screenTop - height;
        return event.getX() >= screenLeft
                && event.getX() <= screenRight
                && event.getY() >= screenBottom
                && event.getY() <= screenTop;
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiListBounds.kt
package weaponsprocurement.gui

import com.fs.starfarer.api.input.InputEventAPI
import com.fs.starfarer.api.ui.CustomPanelAPI

class WimGuiListBounds(
    private val maxScrollOffset: Int,
    private val left: Float,
    private val top: Float,
    private val width: Float,
    private val height: Float,
) {
    fun getMaxScrollOffset(): Int = maxScrollOffset

    fun contains(root: CustomPanelAPI?, event: InputEventAPI?): Boolean {
        if (root == null || root.position == null || event == null || maxScrollOffset <= 0) {
            return false
        }
        val screenLeft = root.position.x + left
        val screenRight = screenLeft + width
        val screenTop = root.position.y + root.position.height - top
        val screenBottom = screenTop - height
        return event.x >= screenLeft &&
            event.x <= screenRight &&
            event.y >= screenBottom &&
            event.y <= screenTop
    }
}
